lambda/putSimDetails.ts

```diff
@@ -4,18 +4,25 @@ import type { SimDetails } from './getSimDetailsFromCache.js'
 
 export const putSimDetails =
 	(db: DynamoDBClient, cacheTableName: string) =>
-	async (
-		iccid: string,
-		simExisting: boolean,
-		simDetails?: SimDetails,
-		ts?: Date,
-	): Promise<void> => {
+	async ({
+		iccid,
+		simExisting,
+		simDetails,
+		historyTs,
+		ts,
+	}: {
+		iccid: string
+		simExisting: boolean
+		simDetails?: SimDetails
+		historyTs?: Date
+		ts?: Date
+	}): Promise<void> => {
 		await db.send(
 			new PutItemCommand({
 				TableName: cacheTableName,
 				Item: marshall({
 					iccid,
-					timestamp: new Date().toISOString(),
+					historyTs: historyTs ? historyTs.toISOString() : '',
```

Comment:
Either use DynamoDB NULL value or omit the value, but do not store a blank string to signal the absence of a value.